sqlmesh/core/config/connection.py

```diff
@@ -269,16 +269,18 @@ def init(cursor: duckdb.DuckDBPyConnection) -> None:
                         query = f"ATTACH '{path_options}'"
                         if not path_options.startswith("md:"):
                             query += f" AS {alias}"
-                        elif self.token:
-                            query += f"?motherduck_token={self.token}"
                     cursor.execute(query)
                 except BinderException as e:
                     # If a user tries to create a catalog pointing at `:memory:` and with the name `memory`
                     # then we don't want to raise since this happens by default. They are just doing this to
                     # set it as the default catalog.
+                    # If a user tried to attach a MotherDuck database/share which has already by attached via
+                    # `ATTACH 'md:'`, then we don't want to raise since this is expected.
                     if not (
                         'database with name "memory" already exists' in str(e)
                         and path_options == ":memory:"
+                    ) and not (
```

Comment:
Currently, the `ATTACH` commands for the individual MotherDuck databases included in `catalogs` will raise an error since they have been automatically attached by the catch all `md:` in the connection string.

We could either:
1. Ignore the errors (this way)
2. Skip `ATTACH`ing the MD catalogs and assume they exist
3. Skip ATTACHing the catalogs but verify they are already attached and available